tests/test_grid.py

```diff
@@ -472,3 +472,31 @@ def test_gaussian_smooth():
     assert uniform.active_scalars_name == active
     assert uniform.active_scalars.shape == values.shape
     assert not np.all(uniform.active_scalars == values)
+
+
+@pytest.mark.parametrize('ind', [range(10), np.arange(10), np.
+                                 ones(beam.n_cells, np.bool)])
+def test_remove_cells(ind):
+    grid_copy = beam.copy()
+    grid_copy.remove_cells(ind)
+    assert grid_copy.n_cells < beam.n_cells
+
+
+def test_remove_cells_not_inplace():
+    grid_copy = beam.copy()  # copy to protect
+    grid_w_removed = grid_copy.remove_cells(range(10), inplace=False)
+    assert grid_w_removed.n_cells < beam.n_cells
+    assert grid_copy.n_cells == beam.n_cells
+
+
+def test_remove_cells_invalid():
+    grid_copy = beam.copy()
+    with pytest.raises(ValueError):
+        grid_copy.remove_cells(np.ones(10, np.bool))
+
+
+def test_hide_cells():
+    sgrid_copy = sgrid.copy()
+    sgrid_copy.hide_cells(range(10, 20))
+    sgrid_copy.UpdateCellGhostArrayCache()
```

Comment:
Why is this necessary, respectively shouldn't this happen in `hide_cells`?

Reply:
The array cache is automatically updated when setting `hide_cells`.  Thanks for catching that.